src/common/lock.c

```diff
@@ -280,3 +287,58 @@ lockRelease(bool failOnNoLock)
 
     FUNCTION_LOG_RETURN(BOOL, result);
 }
+
+/**********************************************************************************************************************************/
+bool
+writeLockPercentageComplete(const String *lockPath, const String *stanza, const String *execId, const double percentageComplete)
+{
+    FUNCTION_LOG_BEGIN(logLevelDebug);
+    FUNCTION_LOG_PARAM(STRING, stanza);
+    FUNCTION_LOG_PARAM(STRING, execId);
+    FUNCTION_LOG_PARAM(DOUBLE, percentageComplete);
+    FUNCTION_LOG_END();
+
+    ASSERT(lockPath != NULL);
+    ASSERT(stanza != NULL);
+    ASSERT(execId != NULL);
+
+    bool status = false;
+
+    // This context should already exist, noop if it doesn't
+    if (lockMemContext != NULL)
+    {
+        MEM_CONTEXT_BEGIN(lockMemContext)
```

Comment:
I don't think any of this needs to be in `lockMemContext`. A temp mem context would be better.

Reply:
updated in 0f56ed01..1f3fbd2b